### DISCUSSION TITLE
Watching for storage change in GUI in addon

### DISCUSSION THREAD

**Discussion by ryncsn** [rhinstaller/anaconda#3508] [Q&A] [answered]
Hi,

I'm working on kdump-anaconda-addon and trying to watch for storage change in GUI. The idea is simple, whenever a storage change is made, kdump-anaconda-addon will check if there is any LUKS encrypted partition. And if there is, it will show a warning message on the GUI hub page. (The error message will gone after the user clicks into the kdump spoke and checked the warning info).

But it seems there is no existing hook/watcher interface for doing this, so I tried to re-detect storage info whenever the ready() method is called for the kdump GUI spoke:

https://github.com/ryncsn/kdump-anaconda-addon/commit/a443a162d55d24b5c3e07556e47db65873c88a68#diff-5854bfbd9b22360e505e10c4e3b3aac8e6aed20a6f2e9cff5b133de39e7aa76cR161

Is this a reliable way to do so? From my side, it just works. But not sure if there is any risk about this implementation.

**Comment by poncovka** [comment, marked as answer]
Hi, you should be able to watch the `AppliedPartitioning` property of the Storage module. It would look like this:
```
# Connect a callback to the PropertiesChanged signal.
storage = STORAGE.get_proxy()
storage.PropertiesChanged.connect(callback)

# Check the changed properties in the callback.
def callback(interface, changed, invalid):
  if changed.get("AppliedPartitioning"):
    print("The partitioning has changed.")
```

The property returns an empty string or a DBus path to a checked and valid partitioning module, that will be used for the installation. This is also the partitioning that you should check for the LUKS devices.

**Reply by ryncsn** [reply]
Thanks! I've updated my code according to your suggestion.
